src/clib/lib/job_queue/torque_driver.cpp

```diff
@@ -845,10 +845,57 @@ job_status_type torque_driver_get_job_status(void *__driver, void *__job) {
 
 void torque_driver_kill_job(void *__driver, void *__job) {
 
+    char *tmp_std_file =
+        (char *)util_alloc_tmp_file("/tmp", "ert-qdel-std", true);
+    char *tmp_err_file =
+        (char *)util_alloc_tmp_file("/tmp", "ert-qdel-err", true);
+
     auto driver = static_cast<torque_driver_type *>(__driver);
     auto job = static_cast<torque_job_type *>(__job);
-    spawn_blocking(driver->qdel_cmd, 1, (const char **)&job->torque_jobnr_char,
-                   NULL, NULL);
+    torque_debug(driver, "Killing Torque job: '%s %s'", driver->qdel_cmd,
+                 job->torque_jobnr_char);
+
+    /* The qdel command might fail intermittently for acceptable reasons,
+           retry a couple of times with exponential sleep. */
+    int return_value = -1;
+    bool qdel_succeeded = false;
+    int retry_interval = 2; /* seconds */
+    int slept_time = 0;
+    while ((return_value != 0) && (slept_time <= driver->timeout)) {
+        return_value = spawn_blocking(driver->qdel_cmd, 1,
+                                      (const char **)&job->torque_jobnr_char,
+                                      tmp_std_file, tmp_err_file);
+        if (return_value != 0) {
+            if (slept_time + retry_interval <= driver->timeout) {
+                torque_debug(driver,
+                             "qdel failed for job %s with exit code "
+                             "%d, retrying in %d seconds",
+                             job->torque_jobnr_char, return_value,
+                             retry_interval);
+                sleep(retry_interval);
```

Comment:
I see that on Torque we have some `usleep(rand() % 2000000); // max 2 seconds` in addition - why is that?; ie. why not here?

Reply:
Comment in the code: "Sleep some more at random, to avoid synchronized retries from all threads". This is relevant for qstat which is ddos'ing the queue server. qdel should not need this drastic measure.